openedx/openedx-platform
# [ENT-173] Remove data sharing consent from logistration

This PR removes data sharing consent from the main OpenEdX logistration form. Logic already exists to ensure that users logging in via SSO are redirected to a consent form when relevant, so further work within the Enterprise app to "catch" the user and get their consent isn't necessary.

**JIRA tickets**: Implements [ENT-173](https://openedx.atlassian.net/browse/ENT-173)

**Sandbox URL**: http://137.74.28.117 (staff/edx)

**Merge deadline**: 28 February 2017

**Testing instructions**:

1. With the sandbox, browse to /admin/enterprise, and verify that a single Enterprise Customer is present, that the Enterprise Customer has an attached SSO provider, and that data sharing consent is disabled for that Enterprise Customer.
2. Register using the TestShib provider.
    1. Observe that no information about data sharing consent is presented.
3. Unlink the user from the SSO provider by going to the Account page and clicking "Unlink" under TestShib.
3. Reconfigure the Enterprise Customer in the admin to enable data sharing consent, and require it at login.
4. Register using the TestShib provider.
    1. Observe that there is no information about data sharing consent on the registration page.
    2. Observe that after registering, you are redirected to a page where you can provide consent.
    3. Observe that if you do not provide consent, you are redirected to the login page.
    4. Repeat, this time consenting, and observing that you are taken to the Dashboard. 
5. As in step 3, unlink the user from the SSO provider.
6. Update the Enterprise Customer's configuration to require consent at course enrollment.
7. Register using the TestShib provider.
    1. Observe that there is no information about data sharing consent on the registration page.
    2. Observe that after registering, you are redirected to a page where you can provide consent.
    3. Observe that if you do not provide consent, you are redirected to the Dashboard.

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
